### PR TITLE
Add support for passing build args to signers

### PR DESCRIPTION
**Pull request by cpuguy83**
I considered extending this to allow this for any kind of frontend (and as such sticking it into the `dalec.Frontend` type), however there really shouldn't be different vars passed to target frontends. Hence why I am restricting this to just signers for now.

If we decide to later we can add this more generically, however it will be much more difficult to take it away if we decide it was a bad idea.

Related to #263

### CODE REVIEW

**Review by pmengelbert — CHANGES_REQUESTED** [review]
This looks good. Just to make sure I understand how dalec will behave after this PR is merged, is the following accurate?

1. Values needed for the signing image can be provided via the `package_config` section, either on the spec level, or on the level of a specific target. (good for packages with both linux + windows targets).
2. These values can be supplied at build-invocation time by leveraging the spec's `args`.
3. The args will be read by the consumer (i.e. the signing image) via the filesystem, and will have to be initialized into the environment on that end.

The only actionable request I have is that this should be captured in an example somewhere (in the `docs/examples` directory perhaps). As we keep adding features like this, having an up-to-date example with the kitchen sink makes it easier for all the developers on the project to track spec changes.

**Comment by cpuguy83** [comment]
Updated with doc updates.

**Comment by pmengelbert** [comment]
I think this needs to be rebased, then it should pass the CI

**Review by pmengelbert — COMMENTED** [review]
LGTM, will wait for rebase + CI passage to approve.

**Comment by cpuguy83** [comment]
This was rebased, FYI.

**Review by pmengelbert — APPROVED** [review]
LGTM, thanks

**Pull request merged**
